### PR TITLE
feat: delete draft document

### LINKED ISSUE #472: Delete a draft document
**Current Behavior:** 
- While the UI allows to delete draft documents, the underlying functionality is missing.

**Desired Feature:** 
- Enable users to delete documents that have not been sent, opened, or signed from the Actions on the documents table

### PR DESCRIPTION
**Pull request by G3root**
fixes #472

[Screencast 2023-10-02 22:02:59.webm](https://github.com/documenso/documenso/assets/84864519/e224d6c6-239b-4a00-9b53-8598d4408786)

### CODE REVIEW

File: packages/trpc/server/document-router/router.ts [hunk 1]
```diff
@@ -76,6 +77,25 @@ export const documentRouter = router({
       }
     }),
 
+  deleteDocument: authenticatedProcedure
+    .input(ZGetDocumentByIdQuerySchema)
```
Comment (dephraiim) [inline packages/trpc/server/document-router/router.ts:81]: Can you create a specific schema for the delete router?

File: apps/web/src/app/(dashboard)/documents/data-table-action-dropdown.tsx [hunk 2]
```diff
@@ -109,6 +118,10 @@ export const DataTableActionDropdown = ({ row }: DataTableActionDropdownProps) =
     window.URL.revokeObjectURL(link.href);
   };
 
+  const handleDelete = async () => {
```
Comment (dephraiim) [inline apps/web/src/app/(dashboard)/documents/data-table-action-dropdown.tsx:121]: Can you rename `handleDelete` to `onDraftDelete` or something instead? 

We always prefer to use the `on<ActionName>` naming convention for event handlers.

**Review by dephraiim — CHANGES_REQUESTED** [review]

**Comment by dephraiim** [comment]
@G3root  Can you also add a dialog to confirm that you want to delete before? To avoid accidental deletion of documents

**Comment by G3root** [comment]
@dephraiim pushed some changes

[Screencast 2023-10-06 17:38:24.webm](https://github.com/documenso/documenso/assets/84864519/7176a14d-1124-4839-a008-5446a1b6f75c)

File: apps/web/src/app/(dashboard)/documents/data-table-action-dropdown.tsx [hunk 3]
```diff
@@ -147,9 +147,11 @@ export const DataTableActionDropdown = ({ row }: DataTableActionDropdownProps) =
           Void
         </DropdownMenuItem>
 
-        <DropdownMenuItem disabled>
-          <Trash2 className="mr-2 h-4 w-4" />
-          Delete
+        <DropdownMenuItem disabled={row.status !== 'DRAFT'} asChild>
+          <Link href={`/documents/delete/${row.id}`}>
```
Comment (dguyen) [inline apps/web/src/app/(dashboard)/documents/data-table-action-dropdown.tsx:151]: Don't need to create a new page, just directly open the dialog.

**Review by dguyen — CHANGES_REQUESTED** [review]

**Comment by dguyen** [comment]
Thanks for adding this feature in @G3root!

/tip 50

**Pull request merged**